Tools/CMake/AMReXInstallHelpers.cmake

```diff
@@ -23,25 +23,28 @@ function ( generate_amrex_config_header )
 
    endforeach ()
 
-   if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" )
-      set(COMPILER_ID_MACRO __GNUC__)
-   elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel" )
-      set(COMPILER_ID_MACRO __INTEL_COMPILER)
-   elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Cray" )
-      set(COMPILER_ID_MACRO  __CRAYC)
-   elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI" )
-      set(COMPILER_ID_MACRO  __PGI)
-   elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" )
-      set(COMPILER_ID_MACRO  __llvm__)
-   elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang" )
-      set(COMPILER_ID_MACRO  __llvm__)
-   elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC" )
-      set(COMPILER_ID_MACRO  _MSC_VER)
-   else ()
-      # avoid malforming the AMReX_Config.H file by setting something valid
-      message(FATAL_ERROR "Compiler '${CMAKE_CXX_COMPILER_ID}' not supported by AMReX developers! "
-                      "Try to configure with -DALLOW_DIFFERENT_COMPILER=ON")
-   endif ()
+   set(COMP_DECLS)
+   if (NOT ALLOW_DIFFERENT_COMPILER)
+       if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" )
+           set(COMPILER_ID_MACRO __GNUC__)
+       elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel" )
+           set(COMPILER_ID_MACRO __INTEL_COMPILER)
+       elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Cray" )
+           set(COMPILER_ID_MACRO  __CRAYC)
+       elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI" )
+           set(COMPILER_ID_MACRO  __PGI)
+       elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" )
+           set(COMPILER_ID_MACRO  __llvm__)
+       elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang" )
+           set(COMPILER_ID_MACRO  __llvm__)
+       elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC" )
+           set(COMPILER_ID_MACRO  _MSC_VER)
+       else ()
+           message(FATAL_ERROR "Compiler '${CMAKE_CXX_COMPILER_ID}' not supported by AMReX developers! "
+               "Try to configure with -DALLOW_DIFFERENT_COMPILER=ON")
+       endif ()
+       set(COMP_DECLS "\n#ifndef ${COMPILER_ID_MACRO}\nstatic_assert(false,\"libamrex was built with ${CMAKE_CXX_COMPILER_ID}\");\n#endif")
```

Comment:
Can we inform the user the allow_different_compiler option in static assert's error message?

Reply:
Sure. Should we change that in configure/GNUMake as well?

Reply:
Actually there is no need to inform the user of the option in the static assert since the configure step will exit with a fatal error if ALLOW_DIFFERENT_COMPILER=off when an unknown compiler is used. This is consistent with what `configure.py` does

Reply:
But if the user is not using cmake to build app but using cmake to build amrex, they will not see it.

If you can modify the `mkconfig.py`, that would be great.  There the message is different. `./configure --allow_different_compiler=yes`.

Reply:
I modified the error message according to your suggestion.

Reply:
Thanks!  Looks like the modification of `mkconfig.py` is not finished yet.

Reply:
Oops. Should be fixed now